proto/data.go

```diff
@@ -321,13 +321,24 @@ func (v *Value) Verify(key []byte) error {
 				cksum, Key(key), v)
 		}
 	}
-	if v.Bytes != nil && v.Integer != nil {
-		return util.Errorf("both the value byte slice and integer fields are set for key %s: [% x]",
-			Key(key), v)
-	}
 	return nil
 }
 
+// SetInteger encodes the specified int64 value into the bytes field of the
+// receiver.
+func (v *Value) SetInteger(i int64) {
+	v.Bytes = encoding.EncodeUint64(nil, uint64(i))
+}
+
+// GetInteger decodes an int64 value from the bytes field of the receiver.
+func (v *Value) GetInteger() int64 {
+	if v == nil || v.Bytes == nil {
+		return 0
+	}
+	_, u := encoding.DecodeUint64(v.Bytes)
```

Comment:
The vast majority of uses cases are in tests (`storage/client_raft_test.go` has 11 uses itself). The real uses are in `client/table.go` and `storage/engine/mvcc.go`. Let me clean this up in another PR.